### PR TITLE
[SYCL][CUDA][HIP] Implement piextUSMEnqueueMemcpy2D

### PR DESCRIPTION
**Pull request by abagusetty**

### CODE REVIEW

**Comment by abagusetty** [comment]
@steffenlarsen Would be great if you too can take a peek

File: sycl/plugins/cuda/pi_cuda.cpp [hunk 1]
```diff
@@ -5354,14 +5354,102 @@ pi_result cuda_piextUSMEnqueueMemset2D(pi_queue, void *, size_t, int, size_t,
   return {};
 }
 
-// TODO: Implement this. Remember to return true for
-//       PI_EXT_ONEAPI_CONTEXT_INFO_USM_MEMCPY2D_SUPPORT when it is implemented.
```
Comment (steffenlarsen) [inline sycl/plugins/cuda/pi_cuda.cpp:5358]: Remember this, otherwise the function won't be called. 😉

File: sycl/plugins/cuda/pi_cuda.cpp [hunk 2]
```diff
@@ -5354,14 +5354,102 @@ pi_result cuda_piextUSMEnqueueMemset2D(pi_queue, void *, size_t, int, size_t,
...
+/// \param num_events_in_waitlist is the number of events to wait on
+/// \param events_waitlist is an array of events to wait on
+/// \param event is the event that represents this operation
+pi_result cuda_piextUSMEnqueueMemcpy2D(pi_queue Queue, pi_bool Blocking,
+                                       void *DstPtr, size_t DstPitch,
+                                       const void *SrcPtr, size_t SrcPitch,
+                                       size_t Width, size_t Height,
+                                       pi_uint32 NumEventsInWaitlist,
+                                       const pi_event *EventWaitlist,
+                                       pi_event *Event) {
```
Comment (steffenlarsen) [inline sycl/plugins/cuda/pi_cuda.cpp:5377]: To keep it consistent with the rest of the file, I would use lowercase letter and _ in argument names. It does not live up to the coding standard, but when we want to address that we should give the file a sweep. Changing this also makes the above documentation accurate.

**Review by steffenlarsen — CHANGES_REQUESTED** [review]
Couple of comments, but otherwise it looks good.

It may be the extension title being confusing, but this is not actually implementing all of sycl_ext_oneapi_memcpy2d but just the piextUSMEnqueueMemcpy2D side. Could you please change the title of this PR to reflect this?

File: sycl/plugins/cuda/pi_cuda.cpp (hunk 1 above)
Comment (abagusetty) [reply sycl/plugins/cuda/pi_cuda.cpp:5358]: https://github.com/intel/llvm/pull/7941/commits/e26ce84887bda00ca5cc8e6caafd062fd414a734

**Review by steffenlarsen — APPROVED** [review]
LGTM! Thanks! 🚀

**Comment by bader** [comment]
@abagusetty, please, fix test failures on CUDA.

**Comment by abagusetty** [comment]
@steffenlarsen @bader Sorry about the delay. Relevant CUDA tests were fixed however ESIMD test is failing which I am not sure.

**Comment by bader** [comment]
Failed Tests (1):
  SYCL :: dword_atomic_smoke.cpp

**Pull request merged**